packages/react-client/src/ReactFlightClient.js

```diff
@@ -1353,6 +1353,26 @@ function waitForReference<T>(
   map: (response: Response, model: any, parentObject: Object, key: string) => T,
   path: Array<string>,
 ): T {
+  if (
+    __DEV__ &&
+    // TODO: This should check for the existence of the "readable" side, not the "writable".
```

Comment:
We never pass anything into the response when there's a readable. We could maybe add a boolean just for this.